test/utils/MockStrategyRecipientsExtension.sol

```diff
@@ -0,0 +1,472 @@
+
+        RecipientInitializeData memory _initializeData = abi.decode(_data, (RecipientInitializeData));
+        __RecipientsExtension_init(_initializeData);
+    }
+
+    function _allocate(address[] memory _recipients, uint256[] memory _amounts, bytes memory _data, address _sender)
+        internal
+        override
+    {}
+
+    function _distribute(address[] memory _recipientIds, bytes memory _data, address _sender) internal override {}
+
+    function mock_call___RecipientsExtension_init(IRecipientsExtension.RecipientInitializeData memory _initializeData)
+        public
+    {
+        vm.mockCall(
+            address(this),
+            abi.encodeWithSignature(
+                "__RecipientsExtension_init(IRecipientsExtension.RecipientInitializeData)", _initializeData
+            ),
+            abi.encode()
+        );
+    }
+
+    function __RecipientsExtension_init(IRecipientsExtension.RecipientInitializeData memory _initializeData)
+        internal
+        override
+    {
+        (bool _success, bytes memory _data) = address(this).call(
+            abi.encodeWithSignature(
+                "__RecipientsExtension_init(IRecipientsExtension.RecipientInitializeData)", _initializeData
+            )
+        );
+
+        if (_success) return abi.decode(_data, ());
+        else return super.__RecipientsExtension_init(_initializeData);
+    }
+
+    function call___RecipientsExtension_init(IRecipientsExtension.RecipientInitializeData memory _initializeData)
+        public
+    {
+        return __RecipientsExtension_init(_initializeData);
+    }
+
+    function expectCall___RecipientsExtension_init(IRecipientsExtension.RecipientInitializeData memory _initializeData)
+        public
+    {
+        vm.expectCall(
+            address(this),
+            abi.encodeWithSignature(
+                "__RecipientsExtension_init(IRecipientsExtension.RecipientInitializeData)", _initializeData
+            )
+        );
+    }
+
+    function mock_call__getRecipientStatus(address _recipientId, IRecipientsExtension.Status _returnParam0) public {
+        vm.mockCall(
+            address(this),
+            abi.encodeWithSignature("_getRecipientStatus(address)", _recipientId),
+            abi.encode(_returnParam0)
+        );
+    }
+
+    function _getRecipientStatus(address _recipientId)
+        internal
+        view
+        override
+        returns (IRecipientsExtension.Status _returnParam0)
+    {
+        (bool _success, bytes memory _data) =
+            address(this).staticcall(abi.encodeWithSignature("_getRecipientStatus(address)", _recipientId));
+
+        if (_success) return abi.decode(_data, (IRecipientsExtension.Status));
+        else return super._getRecipientStatus(_recipientId);
+    }
+
+    function call__getRecipientStatus(address _recipientId)
+        public
+        returns (IRecipientsExtension.Status _returnParam0)
+    {
+        return _getRecipientStatus(_recipientId);
+    }
```

Comment:
I'm not sure I understand these design. What is the purpose of `staticcall` and each function? Won't `success` always be false?

If the idea was to simply [expose](https://book.getfoundry.sh/tutorials/best-practices#internal-functions) the internal function I think it's enough with:
```solidity
function expose_getRecipientStatus(address _recipientId) {
   return _getRecipientStatus(_recipientId);
}
```